Python 3 - resources_task creating unexpected targets after reset

## Problem
Both of the `test_incremental()` tests are failing in `test_resources_task.py`.

```
==================== FAILURES ====================
                     _ MinimalImplResourcesTaskTest.test_incremental __

                     self = <pants_test.backend.jvm.tasks.test_resources_task.MinimalImplResourcesTaskTest testMethod=test_incremental>

                         def test_incremental(self):
                           task = self.create_resources_task(target_roots=[self.create_target('a:1'),
                                                                           self.create_target('b:1')])
                           processed_targets = task.execute()

                           self.assertEqual(sorted([self.target('a:1'), self.target('b:1')]), sorted(processed_targets))
                           self.assert_products(task, self.target('a:1'), 1)
                           self.assert_products(task, self.target('b:1'), 1)

                           # Test incremental works - no changes.
                           self.reset_build_graph()
                           task = self.create_resources_task(target_roots=[self.create_target('a:1'),
                                                                           self.create_target('b:1')])
                           processed_targets = task.execute()

                     >     self.assertEqual([], processed_targets)
                     E     AssertionError: Lists differ: [] != [TestTarget(a:1), TestTarget(b:1)]
                     E
                     E     Second list contains 2 additional elements.
                     E     First extra element 0:
                     E     TestTarget(a:1)
                     E
                     E     - []
                     E     + [TestTarget(a:1), TestTarget(b:1)]

                     .pants.d/pyprep/sources/0149f6d60c1f81255ffbf3cd4ca6bef56274a538/pants_test/backend/jvm/tasks/test_resources_task.py:136: AssertionError
```

## To reproduce
1. `./pants clean-all`
1. `./pants --python-setup-interpreter-constraints='["CPython>=3.5"]' test tests/python/pants_test/backend/jvm/tasks:test_resources_task.py`

## My confusion
I don't understand why `process_targets` is supposed to be empty in the last line, after calling `task.execute()`.

```python
  def test_incremental(self):
    task = self.create_resources_task(target_roots=[self.create_target('a:1'),
                                                    self.create_target('b:1')])
    processed_targets = task.execute()

    self.assertEqual(sorted([self.target('a:1'), self.target('b:1')]), sorted(processed_targets))
    self.assert_products(task, self.target('a:1'), 1)
    self.assert_products(task, self.target('b:1'), 1)

    # Test incremental works - no changes.
    self.reset_build_graph()
    task = self.create_resources_task(target_roots=[self.create_target('a:1'),
                                                    self.create_target('b:1')])
    processed_targets = task.execute()

    self.assertEqual([], processed_targets)
```

--- a/contrib/python/src/python/pants/contrib/python/checks/tasks/python_eval.py
+++ b/contrib/python/src/python/pants/contrib/python/checks/tasks/python_eval.py
@@ -203,7 +203,7 @@ class PythonEval(LintTaskMixin, ResolveRequirementsTaskBase):
     return modules
 
   def _get_executable_file_content(self, exec_pex_parent, modules):
-    generator = Generator(pkgutil.get_data(__name__, self._EVAL_TEMPLATE_PATH),
+    generator = Generator(pkgutil.get_data(__name__, self._EVAL_TEMPLATE_PATH).decode('utf-8'),
                           chroot_parent=exec_pex_parent, modules=modules)
     return generator.render()
 

--- a/contrib/python/tests/python/pants_test/contrib/python/checks/tasks/test_python_eval.py
+++ b/contrib/python/tests/python/pants_test/contrib/python/checks/tasks/test_python_eval.py
@@ -6,8 +6,6 @@ from __future__ import absolute_import, division, print_function, unicode_litera
 
 from textwrap import dedent
 
-import pytest
-from future.utils import PY3
 from pants.backend.python.subsystems.python_repos import PythonRepos
 from pants.backend.python.subsystems.python_setup import PythonSetup
 from pants_test.backend.python.tasks.python_task_test_base import PythonTaskTestBase
@@ -15,8 +13,6 @@ from pants_test.backend.python.tasks.python_task_test_base import PythonTaskTest
 from pants.contrib.python.checks.tasks.python_eval import PythonEval
 
 
-# TODO(python3port): https://github.com/pantsbuild/pants/issues/6354. Fix before switching fully to Py3.
-@pytest.mark.skipif(PY3, reason='PEX issue when using Python 3. https://github.com/pantsbuild/pants/issues/6354')
 class PythonEvalTest(PythonTaskTestBase):
 
   @classmethod

--- a/src/python/pants/invalidation/build_invalidator.py
+++ b/src/python/pants/invalidation/build_invalidator.py
@@ -223,7 +223,7 @@ class BuildInvalidator(object):
   def _read_sha_by_id(self, id):
     try:
       with open(self._sha_file_by_id(id), 'rb') as fd:
-        return fd.read().strip()
+        return fd.read().strip().decode('utf-8')
     except IOError as e:
       if e.errno != errno.ENOENT:
         raise

